internal/preload/instance_type.go

```diff
@@ -0,0 +1,89 @@
+package preload
+
+import (
+	"fmt"
+
+	"github.com/RHEnVision/provisioning-backend/internal/clients"
+	"github.com/RHEnVision/provisioning-backend/internal/middleware"
+)
+
+type instanceType struct {
+	filename string
+	path     string
+	etagName string
+	tag      *middleware.ETag
+	typeInfo clients.InstanceTypeInfo
+}
+
+func (p *instanceType) Load() error {
+	// load instance types
+	typesBuf, err := fsTypes.ReadFile(p.filename)
+	if err != nil {
+		return fmt.Errorf("unable to read instance types %s: %w", p.filename, err)
+	}
+
+	err = p.typeInfo.RegisteredTypes.Load(typesBuf)
```

Comment:
Just wondering here: 
Isn't it a problem that the RegisteredTypes is nil? in `ec2_types.go` we are not initializing the typeInfo field.

Reply:
It is not, `Load` calls `yaml.Unmarshal` which apparently works with `nil` values. In fact, since you pass a pointer, it is actually not important what value it contains because it will be overwritten and garbage collected.